src/main/java/core/utils/CredentialDeserializer.java

```diff
@@ -0,0 +1,78 @@
+package core.utils;
+
+import clickup.ui.entities.User;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.apache.log4j.spi.ErrorCode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+
+/**
+ * Deserializes the credentials information provided as a json file to a POJO entity class.
+ *
+ * @author Alejandro Sánchez Luizaga
+ * @version 1.0
+ */
+public final class CredentialDeserializer {
+    private static CredentialDeserializer instance;
+    private static JsonObject credentials;
+    private static final String USER_RSRC_CONFIG_FILE = "user.properties";
+    private static final Logger LOGGER = LoggerFactory.getLogger(CredentialDeserializer.class);
+
+    /**
+     * Utility class private constructor.
+     */
+    private CredentialDeserializer() {
+        initialize();
+    }
+
+    /**
+     * Returns the same and only instance of this class.
+     *
+     * @return the same instance of CredentialDeserializer class.
+     */
+    public static CredentialDeserializer getInstance() {
+        if ((instance == null) || (credentials == null)) {
+            instance = new CredentialDeserializer();
+        }
+        return instance;
+    }
+
+    /**
+     * Deserializes the credentials information provided as a json file to a POJO entity class.
+     *
+     * @param userType a String containing the type of user to retrieve, i.e. user or admin.
+     * @return a POJO entity instanceaccording to the value set for userType in the resource file.
+     */
+    public static User getUser(final String userType) {
+        PropertyReader.loadFile(USER_RSRC_CONFIG_FILE);
+        String user = PropertyReader.retrieveField(userType);
+        JsonElement jsonUser = credentials.getAsJsonObject(userType).getAsJsonObject(user);
+        Gson gson = new Gson();
+        User userObject = gson.fromJson(jsonUser, User.class);
+        return userObject;
+    }
+
+    /**
+     * Sets the initial value of credentials attribute of this class.
+     */
+    private void initialize() {
+        JsonParser parser = new JsonParser();
+        JsonElement jsonElement = null;
+        PropertyReader.loadFile(USER_RSRC_CONFIG_FILE);
+        String jsonDirectory = PropertyReader.retrieveField("dir.json");
+        try {
+            jsonElement = parser.parse(new FileReader(jsonDirectory.concat(System.getProperty("file.separator"))
```

Comment:
Maybe this part when you use "file.separator" it could be better declare as  a  constant. What do you thing?

Reply:
"file.separator" is a Java language identificator.